packages/clerk-js/src/ui/common/SSOCallback.tsx

```diff
@@ -17,16 +17,18 @@ export const SSOCallback = withCardStateProvider<HandleOAuthCallbackParams | Han
 });
 
 export const SSOCallbackCard = (props: HandleOAuthCallbackParams | HandleSamlCallbackParams) => {
-  const { handleRedirectCallback } = useClerk();
+  const { handleRedirectCallback, __internal_setActiveInProgress } = useClerk();
   const { navigate } = useRouter();
   const card = useCardState();
 
   React.useEffect(() => {
     let timeoutId: ReturnType<typeof setTimeout>;
-    handleRedirectCallback({ ...props }, navigate).catch(e => {
-      handleError(e, [], card.setError);
-      timeoutId = setTimeout(() => void navigate('../'), 4000);
-    });
+    if (__internal_setActiveInProgress !== true) {
```

Comment:
❓ why don't we do this in our router instead?

Reply:
We can definitely explore something like that, which I feel is more intelligent than this blunt approach. It'd likely use the same flag too, so we're good there. The main reason I went with this instead of scoping this purely to preventing navigations is I'm concerned we have more functionality that needs to be prevented from executing a second time. For example, the PR as is prevents the execution of `handleRedirectCallback`, rather than simply preventing a navigation from it. `handleRedirectCallback` has several non-navigation actions that it performs that it would be best if they were prevented completely (stuff like `signIn.reload`, `signIn.create`, `signUp.create`, `setActive`, etc).